### PR TITLE
Add expectations for flaky test issue assignments

### PR DESCRIPTION
**Pull request by lavalamp**

### CODE REVIEW

File: docs/devel/flaky-tests.md [hunk 1]
```diff
@@ -32,7 +32,73 @@ Documentation for other releases can be found at
...
+$ godep go test -c -race
+```
+
+Then run it under stress
+
+```
+$ stress ./package.test -test.run=FlakyTest
+```
+
+It runs the command and writes output to `/tmp/gostress-*` files when it fails. It periodically reports with run counts.
```
Comment (krousey) [inline docs/devel/flaky-tests.md]: Is it worth noting that doing this with tests that use the httptest package can easily exhaust the ports on your machine?

File: docs/devel/flaky-tests.md [hunk 2]
```diff
@@ -32,7 +32,73 @@ Documentation for other releases can be found at
...
+you're part of a group that has opted in. We are more than happy to accept help
+from anyone in fixing these, but due to the severity of the problem when merges
+are blocked, we need reasonably quick turn-around time on test flakes. Therefore
+we have the following guidelines:
+
+1. If a flaky test is assigned to you, it's more important than anything else
+   you're doing unless you can get a special dispensation (in which case it will
+   be reassigned).  If you have too many flaky tests assigned to you, or you
+   have such a dispensation, then it's *still* your responsibility to find new
+   owners (this may just mean giving stuff back to the relevant TL).
```
Comment (sarahnovotny) [inline docs/devel/flaky-tests.md]: I think this should be Team Lead or SIG leader... or is TL a designation that exists broadly in the community as well?

**Comment by lavalamp** [comment]
Comments addressed.

File: docs/devel/flaky-tests.md [hunk 3]
```diff
@@ -32,7 +32,76 @@ Documentation for other releases can be found at
...
+
+Therefore, it's very important that we write tests defensively. Situations that
+"almost never happen" happen with some regularity when run thousands of times in
+resource-constrained environments. Since flakes can often be quite hard to
+reproduce while still being common enough to block merges occasionally, it's
+additionally important that the test logs be useful for narrowing down exactly
+what caused the failure.
+
+Note that flakes can occur in unit tests, integration tests, or end-to-end
+tests, but probably occurr most commonly in end-to-end tests.
```
Comment (brendandburns) [inline docs/devel/flaky-tests.md]: s/ocurr/occur/

**Comment by brendandburns** [comment]
one minor typo.  Feel free to self-lgtm once its addressed.

**Comment by lavalamp** [comment]
Fixed typo. I also added a "how to file issues" section.

File: docs/devel/flaky-tests.md [hunk 4]
```diff
@@ -32,7 +32,92 @@ Documentation for other releases can be found at
...
+
+1. Search for the test name. If you find an open issue and you're 90% sure the
+   flake is exactly the same, add a comment instead of making a new issue.
+2. If you make a new issue, you should title it with the test name, prefixed by
+   "e2e/unit/integration flake:" (whichever is appropriate)
+3. Reference any old issues you found in step one.
+4. Paste, in block quotes, the entire log of the individual failing test, not
+   just the failure line.
+5. Link to durable storage with the rest of the logs. This means (for all the
+   tests that Google runs) the GCS link is mandatory! The Jenkins test result
```
Comment (ihmccreery) [inline docs/devel/flaky-tests.md]: @sarahnovotny @thockin @ixdy We should make this easier to do via Jenkins.  This is also crucial if non-Googlers want to help out; Jenkins links are internal-only, but GCS links aren't.

Comment (lavalamp) [inline docs/devel/flaky-tests.md]: Good point, I added it and re-pushed.

**Comment by brendandburns** [comment]
LGTM

**Pull request merged**